org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.java

```diff
@@ -0,0 +1,489 @@
+		if (e != null) {
+			source = e.getSource();
+		}
+
+		if (createUiProject.getSelection() && !isSelected(sourceLayout, SourceLayout.PLAIN)) {
+			if (createUiProject == source) {
+				String message = "'" + createUiProject.getText() + "' requires " + SourceLayout.PLAIN + " source layout. " //
+						+ "Please <a>select '" + SourceLayout.PLAIN + "'</a> source layout.";
+				reportIssue(ERROR, message, () -> select(sourceLayout, SourceLayout.PLAIN));
+			} else {
+				String message = SourceLayout.MAVEN + " source layout is not supported by the '" + createUiProject.getText() + "' project. " //
+						+ "Please <a>deselect '" + createUiProject.getText() + "'</a>.";
+				reportIssue(ERROR, message, () -> createUiProject.setSelection(false));
+			}
+		}
+
+		if (isSelected(preferredBuildSystem, BuildSystem.NONE) && isSelected(sourceLayout, SourceLayout.MAVEN)) {
+			if (preferredBuildSystem == source) {
+				String message = "Maven/Gradle source layout is not supported without a build system. " //
+						+ "Please <a>select '" + SourceLayout.PLAIN + "'</a> source layout.";
+				reportIssue(ERROR, message, () -> select(sourceLayout, SourceLayout.PLAIN));
+			} else {
+				String message = "Maven/Gradle source layout is only supported when using Maven or Gradle build system. " //
+						+ "You need to choose Maven or Gradle build system. Select <a>Gradle</a> build.";
+				reportIssue(ERROR, message, () -> select(preferredBuildSystem, BuildSystem.GRADLE));
+			}
+		}
+
+		if (createWebProject.getSelection() && isSelected(preferredBuildSystem, BuildSystem.NONE)) {
+			if (preferredBuildSystem == source) {
+				String message = "The '" + createWebProject.getText() + "' project can not be build without a build system. " //
+						+ "Please <a>deselect '" + createWebProject.getText() + "'</a>.";
+				reportIssue(ERROR, message, () -> createWebProject.setSelection(false));
+			} else {
+				String message = "To build the '" + createWebProject.getText() + "' " //
+						+ "project, you need to choose Maven or Gradle build system." //
+						+ "Select <a>Gradle</a> build.";
+				reportIssue(ERROR, message, () -> select(preferredBuildSystem, BuildSystem.GRADLE));
+			}
+		}
+
+		boolean isSelected = Iterables.any(uiButtons, uiButton -> uiButton.getSelection());
+		if (isSelected) {
+			createIdeProject.setEnabled(false);
+		}
+
+		if (autoSelectIdeProject) {
+			autoSelectIdeProject = false;
+			String message = "'" + createIdeProject.getText() + "' project was automatically selected as option '"
+					+ ((Button) source).getText() + "' requires it.";
+			reportIssue(INFORMATION, message);
+		}
+		
+		if (autoSelectSDKProject) {
+			autoSelectSDKProject = false;
+			String message = "'" + createSDKProject.getText() + "' was automatically selected as option '" //
+					+ ((Button) source).getText() + "' requires it.";
+			reportIssue(INFORMATION, message);
+		}
+	}
+
+	protected void select(Combo combo, Enum<?> enu) {
+		String[] items = combo.getItems();
```

Comment:
Did you test this method? It's different than in Xtend. Don't have an IDE handy to tell me if it's correct

Reply:
Yes, I have tested this method.